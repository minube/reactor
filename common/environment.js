import { DeviceInfo, Platform } from 'react-native';
import PKG from '../package.json';

const { NODE_ENV, REACT_APP_ENV = 'development' } = process.env;
const IS_SERVER = typeof window === 'undefined';
const IS_WEB = Platform.OS === 'web';
const IS_PRODUCTION = NODE_ENV === 'production' && NODE_ENV === REACT_APP_ENV;
const IS_MOBILE_WEB = IS_WEB && !IS_SERVER
  ? /iPhone|Android|Tablet|iPad|Mobile|iPod|Windows Phone|Blackberry/i.test(DeviceInfo.userAgent)
  : false;

if (IS_WEB && !IS_SERVER) window.reactor = { version: PKG.version };

export default {
  IS_NATIVE: !IS_WEB,
  IS_SERVER: IS_SERVER && IS_WEB,
  IS_WEB,
  IS_MOBILE_WEB,
  IS_PRODUCTION,
  IS_DEVELOPMENT: !IS_PRODUCTION,
  IS_TEST: NODE_ENV === 'test',

  NODE_ENV,

  PKG: {
    name: PKG.name,
    version: PKG.version,
  },

  REACT_APP_ENV,
};
