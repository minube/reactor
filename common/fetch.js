const DEFAULT_HEADERS = {
  'Content-Type': 'application/json',
  'X-Requested-With': 'XMLHttpRequest',
};

export default async (endpoint, props = {}) => {
  const { headers, method = 'GET', ...others } = props;

  return new Promise((resolve, reject) => {
    fetch(endpoint, {
      headers: { ...DEFAULT_HEADERS, ...headers },
      method,
      ...others,
    })
      .then(async (response) => {
        const json = await response.json();

        if (response.status >= 400) reject({ code: response.status, message: json.message }); // eslint-disable-line
        else resolve(json);
      }).catch(({ message = 'ERROR_CONNECTION', response } = {}) => {
        reject({ // eslint-disable-line
          code: response ? response.status : undefined,
          message,
        });
      });
  });
};
