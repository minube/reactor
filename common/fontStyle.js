import screenSize from './screenSize';
import THEME from './theme';

const { FONT, UNIT } = THEME;

export default (
  width,
  {
    tiny, small, large, title,
  },
) => {
  const {
    TINY, PHONE, TABLET, SMALL, REGULAR, LARGE,
  } = screenSize(width);

  let fontSize = REGULAR || LARGE ? FONT.SIZE.REGULAR : FONT.SIZE.SMALL;

  if (tiny) fontSize = REGULAR || LARGE ? FONT.SIZE.TINY : UNIT * 1.1;
  else if (small) fontSize = REGULAR || LARGE ? FONT.SIZE.SMALL : FONT.SIZE.TINY;
  else if (large) {
    fontSize = FONT.SIZE.LARGE;
    if (TINY) fontSize = FONT.SIZE.SMALL;
    if (PHONE || SMALL) fontSize = FONT.SIZE.REGULAR;
    if (TABLET || REGULAR) fontSize = UNIT * 1.8;
  } else if (title) {
    fontSize = UNIT * 4.8;
    if (TINY || PHONE || TABLET) fontSize = UNIT * 2.4;
    if (SMALL || REGULAR) fontSize = UNIT * 3;
  }

  return {
    fontSize,
    lineHeight: fontSize * (TINY || SMALL ? 1.2 : 1.3),
  };
};
