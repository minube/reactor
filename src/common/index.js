import C from './constants';
import STYLE from './style';
import THEME from './theme';

export {
  C,
  STYLE,
  THEME,
};
