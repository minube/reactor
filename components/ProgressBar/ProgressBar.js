import React from 'react';
import { Platform, ProgressBar, StyleSheet, View } from 'react-native';

import { THEME } from '../../common';
import styles from './ProgressBar.style';

const { COLOR } = THEME;

export default ({ ...inherit }) => (
  Platform.OS === 'web'
    ? <ProgressBar color={COLOR.PRIMARY} trackColor={COLOR.BACKGROUND} {...inherit} />
    :
    <View style={styles.container}>
      <View style={StyleSheet.create([styles.value, { width: `${(inherit.progress || 0) * 100}%` }])} />
    </View>
);
