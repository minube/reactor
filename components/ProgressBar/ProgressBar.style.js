import { StyleSheet } from 'react-native';

import { THEME } from '../../common';

const { UNIT } = THEME;

export default StyleSheet.create({
  container: {
    width: '100%',
    height: UNIT / 2,
    overflow: 'hidden',
  },

  progress: {
    height: '100%',
  },
});
