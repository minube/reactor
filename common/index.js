import fetch from './fetch';
import LAYOUT from './layout';
import SHAPE from './shape';
import STYLE from './style';
import THEME from './theme';

export {
  fetch,
  LAYOUT,
  SHAPE,
  STYLE,
  THEME,
};
