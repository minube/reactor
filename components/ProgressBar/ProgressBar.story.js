import React from 'react';
import { storiesOf } from '@storybook/react';
import { boolean, color, number } from '@storybook/addon-knobs/react';

import { THEME } from '../../common';
import ProgressBar from './ProgressBar';

storiesOf('✅ ProgressBar', module)
  .addWithJSX('default', () => (
    <ProgressBar />
  ))
  .addWithJSX('progress', () => (
    <ProgressBar progress={0.33} />
  ))
  .addWithJSX('color', () => (
    <ProgressBar progress={0.5} color={THEME.COLOR.ACCENT} />
  ))
  .addWithJSX('trackColor', () => (
    <ProgressBar progress={0.5} trackColor={THEME.COLOR.ERROR} />
  ))
  .addWithJSX('style', () => (
    <ProgressBar progress={0.5} style={{ borderRadius: THEME.OFFSET / 2, height: THEME.OFFSET }} />
  ))
  .addWithJSX('🏀 Playground', () => (
    <ProgressBar
      progress={number('progress', 0.5)}
      indeterminate={boolean('indeterminate', false)}
      color={color('color', THEME.COLOR.ACCENT)}
      trackColor={color('trackColor', THEME.COLOR.BACKGROUND)}
    />
  ));
