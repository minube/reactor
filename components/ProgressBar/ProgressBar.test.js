import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import { THEME } from '../../common';
import ProgressBar from './ProgressBar';

describe('<ProgressBar>', () => {
  it('renders', () => {
    const tree = renderer.create(<ProgressBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {progress}', () => {
    const tree = renderer.create(<ProgressBar progress={0.33} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {color}', () => {
    const tree = renderer.create(<ProgressBar color={THEME.COLOR.ACCENT} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {trackColor}', () => {
    const tree = renderer.create(<ProgressBar trackColor={THEME.COLOR.ACCENT} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {style}', () => {
    const tree = renderer.create(<ProgressBar style={{ borderRadius: THEME.OFFSET / 2, height: THEME.OFFSET }} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders in DOM environment', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ProgressBar />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});

