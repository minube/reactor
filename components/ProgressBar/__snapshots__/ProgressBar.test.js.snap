// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressBar> renders 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "backgroundColor": "#f4f4f4",
      "height": 5,
      "overflow": "hidden",
      "width": "100%",
    }
  }
>
  <View
    collapsable={undefined}
    style={
      Object {
        "backgroundColor": "#10c0bc",
        "height": "100%",
        "width": 0,
      }
    }
  />
</View>
`;

exports[`<ProgressBar> when {color} 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "backgroundColor": "#f4f4f4",
      "height": 5,
      "overflow": "hidden",
      "width": "100%",
    }
  }
>
  <View
    collapsable={undefined}
    style={
      Object {
        "backgroundColor": "#f69a37",
        "height": "100%",
        "width": 0,
      }
    }
  />
</View>
`;

exports[`<ProgressBar> when {progress} 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "backgroundColor": "#f4f4f4",
      "height": 5,
      "overflow": "hidden",
      "width": "100%",
    }
  }
>
  <View
    collapsable={undefined}
    style={
      Object {
        "backgroundColor": "#10c0bc",
        "height": "100%",
        "width": 0,
      }
    }
  />
</View>
`;

exports[`<ProgressBar> when {style} 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "backgroundColor": "#f4f4f4",
      "borderRadius": 8,
      "height": 16,
      "overflow": "hidden",
      "width": "100%",
    }
  }
>
  <View
    collapsable={undefined}
    style={
      Object {
        "backgroundColor": "#10c0bc",
        "height": "100%",
        "width": 0,
      }
    }
  />
</View>
`;

exports[`<ProgressBar> when {trackColor} 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "backgroundColor": "#f69a37",
      "height": 5,
      "overflow": "hidden",
      "width": "100%",
    }
  }
>
  <View
    collapsable={undefined}
    style={
      Object {
        "backgroundColor": "#10c0bc",
        "height": "100%",
        "width": 0,
      }
    }
  />
</View>
`;
